src/app-layout/__integ__/app-layout-split-panel-forced-position.test.ts

```diff
@@ -0,0 +1,129 @@
+      scrollbarDisplayCondition,
+    });
+
+    const page = new AppLayoutSplitViewPage(browser);
+    await page.setWindowSize({ ...viewports.desktop, width: initialViewportWidth });
+
+    const url = `#/light/app-layout/with-split-panel-and-body-scroll?${params.toString()}`;
+
+    await browser.url(url);
+    const content = createWrapper().findAppLayout().findContentRegion();
+    await page.waitForVisible(content.toSelector());
+    await testFn(page);
+  });
+
+describe('Split panel forced position', () => {
+  describe.each(['classic', 'refresh', 'refresh-toolbar'] as const)('%s', theme => {
+    // Viewport width which makes the split panel switch to the bottom
+    const splitPanelBreakpoints = {
+      classic: 979,
+      refresh: 1030,
+      'refresh-toolbar': 907,
+    };
+
+    const scrollbarThickness = 15;
+
+    const narrow = splitPanelBreakpoints[theme];
+
+    // Account 2 times for scrollbarThickness:
+    // once for the threshold we give to the logic to decide when to transition to the side,
+    // and once for the cases where there is a "real" scrollbar taking up actual space.
+    const wide = narrow + scrollbarThickness * 2 + 1;
+
+    const cases: [
+      string,
+      { scrollbarDisplayCondition: string; scrollbarWhenOnSide: boolean; scrollbarWhenOnBottom: boolean },
+    ][] = [
+      [
+        'without page-level scrollbar',
+        {
+          scrollbarDisplayCondition: 'never',
+          scrollbarWhenOnSide: false,
+          scrollbarWhenOnBottom: false,
+        },
+      ],
+      [
+        'with page-level scrollbar',
+        {
+          scrollbarDisplayCondition: 'always',
+          scrollbarWhenOnSide: true,
+          scrollbarWhenOnBottom: true,
+        },
+      ],
+      [
+        'with page-level scrollbar appearing only when split panel is at the bottom',
+        {
+          scrollbarDisplayCondition: 'bottom',
+          scrollbarWhenOnSide: false,
+          scrollbarWhenOnBottom: true,
+        },
+      ],
+      [
+        'with page-level scrollbar appearing only when split panel is on the side',
+        {
+          scrollbarDisplayCondition: 'side',
+          scrollbarWhenOnSide: true,
+          scrollbarWhenOnBottom: false,
+        },
+      ],
+    ];
+
+    describe.each(cases)(
+      '%s',
+      (_description, { scrollbarWhenOnSide, scrollbarWhenOnBottom, scrollbarDisplayCondition }) => {
+        test(
+          'transitions from side to bottom',
+          setupTest({ scrollbarDisplayCondition, theme, initialViewportWidth: wide }, async page => {
+            // Open split panel and ensure it is on the side.
+            await expect(page.verifySplitPanelPosition('side')).resolves.toBe(true);
+            await expect(page.hasPageScrollbar()).resolves.toBe(scrollbarWhenOnSide);
+
+            // Split panel transitions to the bottom when not enough horizontal space.
+            for (let newWidth = wide; newWidth >= narrow; newWidth--) {
```

Comment:
Why not just `setWindowSize` immediately to `narrow` width? What is the purpose of this loop?

Reply:
The purpose of these loops is to catch eventual crashes due to infinite render loops, as reported in `AWSUI-59954` (page-level scrollbar appears only when split panel is at the bottom), whenever they might happen around the "nominal" breakpoint. If changing the viewport width just once directly from e.g, 270px to 290px, we might not catch the bug, which might happen somewhere in between. And if figuring out what the exact point of failure is, this value might change in the future, opening the door to the bug again.

Unfortunately though, I cannot find a way to catch the other form of the issue (`AWSUI-54993`, page-level scrollbar appears only when split panel is on the side), which manifests itself with flickering instead of a crash. So to your question,

> what happens with the new integration tests if we remove the logic that handles this behavior?

these tests pass, so technically they don't cover the changes in this PR, but they cover the crash case which was present before reverting https://github.com/cloudscape-design/components/pull/2794. And in fact the flickering itself was never really covered, I just verified that the tests https://github.com/cloudscape-design/components/pull/2794 also passed when the other changes in that PR are reverted.